packages/nimble-components/src/table/specs/table-columns-hld.md

```diff
@@ -0,0 +1,256 @@
+    ...
+
+    @attr
+    public valueKey: string;
+
+    @attr
+    public unitKey: string;
+
+    public getDataKeys(): string[] {
+        return [valueKey, unitKey];
+    }
+
+    public cellTemplateFn(): ViewTemplate<ITableCellData> {
+        return html<ITableCellData>`
+            <nimble-text-field readonly="true" value=${x => this.formatData(x.data[this.valueKey], x.data[this.unitKey])}>
+            </nimble-text-field>
+        `;
+    }
+
+    public formatData(value: unknown, unit: unknown): string {
+        return `${(value as string)}, ${(unit as string)}`;
+    }
+}
+```
+
+Here we have a column visualize in different ways based on custom logic:
+
+```TS
+public TableColumnPositiveNegativeNumber extends FoundationElement implements ITableColumn {
+    ...
+
+    @attr
+    public valueKey: string;
+
+    public getDataKeys(): string[] {
+        return [valueKey];
+    }
+
+    public cellTemplateFn(): ViewTemplate<ITableCellData> {
+        return html<ITableCellData>`
+            ${when(x => !this.isPositive(x.data[this.valueKey])),
+            html`
+                <nimble-text-field readonly="true" value=${x => x.data[this.valueKey]} style="fontColor: red;">
+                </nimble-text-field>
+            `}
+            ${when(x => this.isPositive(x.data[this.valueKey])),
+            html`
+                <nimble-text-field readonly="true" value=${x => x.data[this.valueKey]} style="fontColor: green;">
+                </nimble-text-field>
+            `}
+        `;
+    }
+
+    public isPositive(value: unknown): bool {
+        return (value as number) >= 0;
+    }
+}
+```
+
+Finally, here is a column that allows a user to register a callback for a click event on a button inside the cell template:
+
+```TS
+public TableColumnButton extends FoundationElement implements ITableColumn {
+    ...
+
+    @attr
+    public idKey: string;
+
+    public getDataKeys(): string[] {
+        return [valueKey];
+    }
+
+    public callback: (id: string) => void;
+
+    public cellTemplateFn(): ViewTemplate<ITableCellData> {
+        return html<ITableCellData>`
+                <nimble-button readonly="true" @click="${x => this.callback(x.data[this.idKey])}>
+                    <span>Do Something</span>
+                </nimble-button>
+        `;
+    }
+}
+
+Angular template:
+
+<nimble-table>
+    <nimble-table-column-button #tableButton>Press Me</nimble-table-column-button>
+</nimble-table>
+
+Angular code:
+
+@ViewChild('tableButton', { read: NimbleTableColumnButton, static: true }) private readonly tableButton!: NimbleTableColumnButton;
+
+public ngOnInit(): void {
+    tableButton.callback = this.doSomething;
+}
+
+private doSomething(id: string): void {
+    ...
+}
+
+```
+
+Note the missing implementation in the above `ITableColumn` implementations are the necessary pieces to register them as FAST components.
+
+### Initial Nimble-provided Columns
+
+`TableColumnText`
```

Comment:
Do you anticipate having another document that covers the details of the `TableColumnText` component? Or, do you see the approval of this spec being sign off on creating that component as well?

If there is no other spec, I wonder what other information needs to be included here. Some things that immediately come to mind are:
- the element name (e.g. `nimble-table-column-text`)
- explicitly defining what it means that "no value is present" to use the placeholder
    - I assume you mean the value associated with `dataKey` is `null` or `undefined`, but it would be good to solidify this.
- is there a way for a client to provide styling (such as italicizing) the placeholder? Is this something we'd want to do or do the designers have a strong feeling on the way placeholders should always be shown?